boa/src/syntax/ast/node/mod.rs

```diff
@@ -60,6 +60,7 @@ use std::{
 #[cfg(feature = "deser")]
 use serde::{Deserialize, Serialize};
 
+// TODO: This should be split into Expression and Statement.
```

Comment:
I would like to know more about this, given I'm working on changing everything about a  `Node` with spans.

I'm also thinking on creating a `LabelledStatement` node, and it's execution would simply push/pop a label in a "label stack" in the context.

Reply:
> I would like to know more about this, given I'm working on changing everything about a `Node` with spans.

Currently `Node` is both a expression and a statement, it would be better if `Node` was split into how the spec describes it `Expression` , `Statement` (and also `Declaration`). Why do we want this? Right now we don't differentiate between statements (like `if`, `while`, etc) and expressions (like `2 + 2`), while we could make it work like this, it makes somethings awkward like `if (cond) {}` `cond` should **only** and expression, and when we parse it we check for this, but some checks can be elided if we leverage the type system, making illegal state unrepresentable.

The AST should look something like this https://astexplorer.net/ (turn )

Reply:
Makes sense, test but maybe in a different PR

Reply:
Created issue for this https://github.com/boa-dev/boa/issues/1371 :)